### PR TITLE
Fixes savesate for usbmic.

### LINKED ISSUE #13584: [Save state] Loading a microphone app crashes the emulator
### What happens?
Load a microphone app (Audio Mechanica), create a save state, exit to menu, load the app again and then load the save state. The emulator crashes. 

### What *should* happen?
No crash.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

- ppsspp-v1.10.3-1003-g254c316b1-windows-amd64
- Windows 10
- Intel HD3000, DX11

Video : [github](https://github.com/hrydgard/ppsspp/files/5432715/Video.zip) / [imgur](https://i.imgur.com/RHvbbC3.mp4)

### PR DESCRIPTION
**Pull request by shenweip**
Prevents audiobuf from being null pointer. Fixes #13584 .

### CODE REVIEW

File: Core/HLE/sceUsbMic.cpp [hunk 1]
```diff
@@ -128,18 +128,30 @@ void __UsbMicDoState(PointerWrap &p) {
...
+		CoreTiming::RestoreRegisterEvent(eventUsbMicAudioUpdate, "UsbMicAudioUpdate", &__UsbMicAudioUpdate);
+	} else {
+		eventUsbMicAudioUpdate = -1;
+	}
+
+	if (!audioBuf && numNeedSamples > 0) {
+		audioBuf = new QueueBuf(numNeedSamples << 1);
+	}
+	if(eventUsbMicAudioUpdate = -1)
+		eventUsbMicAudioUpdate = CoreTiming::RegisterEvent("UsbMicAudioUpdate", &__UsbMicAudioUpdate);
```
Comment (unknownbrackets) [inline Core/HLE/sceUsbMic.cpp:142]: You can't call RegisterEvent during DoState, it'll potentially return an id that will get allocated by something else later.

-[Unknown]

Comment (shenweip) [reply Core/HLE/sceUsbMic.cpp:142]: Is there a better place to register events for old savestate?

Comment (hrydgard) [reply Core/HLE/sceUsbMic.cpp:142]: Oh right. Read too fast. You need to use RestoreRegisterEvent, see for example sceGe DoSTate.

Comment (hrydgard) [reply Core/HLE/sceUsbMic.cpp:142]: And we need to improve the internal APIs to make this kind of mistake harder to make...

**Pull request merged**